bliss-app/src/components/bliss/bliss-index.jsx

```diff
@@ -3,95 +3,125 @@ import BlissList from "./bliss-list";
 import { useDispatch, useSelector } from "react-redux";
 import {
-	allBliss,
-	blissError,
-	blissStatus,
-	cleanBliss,
-	fetchBliss,
-	prevPage,
-	nextPage,
+  allBliss,
+  blissError,
+  blissStatus,
+  cleanBliss,
+  fetchBliss,
+  prevPage,
+  nextPage,
 } from "../../store/blissSlice";
 import { useSearchParams } from "react-router-dom";
 
-function BlissPage() {
-	const dispatch = useDispatch();
-	const [qparams, setqparams] = useSearchParams();
-	const [limit, setLimit] = useState(15);
-	const [page, setPage] = useState(parseInt(qparams.get("page")));
-	const bliss = useSelector(allBliss);
-	const status = useSelector(blissStatus);
-	const error = useSelector(blissError);
-	const isMounted = useRef(false);
-	let content;
-	const [addBliss, setAddBliss] = useState(false);
+/**
+ * Parse a valid page number from a URLSearchParams object.
+ *
+ * Invalid states, empty states, and NaN are treated as the first page of results.
+ */
+const parsePageFromQueryParams = (query) => {
+  let initialPage = 1;
+  const pageFromQuery = query.get("page");
+  if (pageFromQuery) {
+    initialPage = parseInt(pageFromQuery);
+    if (isNaN(initialPage)) {
+      initialPage = 1;
+    }
+  }
+
+  return initialPage;
+};
+
+/**
+ * Puts page params logic into a hook.
+ *
+ * This will re-run when the query parameters change.
+ *
+ * It returns functions that will change the query parameters to update the page correctly, as well.
+ */
+const usePagination = () => {
+  const limit = 15;
+  const [qparams, setqparams] = useSearchParams();
+
+  const page = parsePageFromQueryParams(qparams);
+
+  const onClickNext = () => {
+    setqparams({ ...qparams, page: page + 1 });
+  };
 
-	const onNextPageClick = async (e) => {
-		e.preventDefault();
-		setPage((page) => page + 1);
-		console.log(page);
-		setqparams({ page });
-		// await dispatch(fetchBliss({limit, offset: page.current === 1 ? 0 : (page.current * limit) - 1, page}))
-	};
+  const onClickPrev = () => {
+    setqparams({ ...qparams, page: page - 1 });
+  };
+
+  // Page = 1: (1 - 1) * 15 = 0 * 15 = 0
+  // Page = 2: (2 - 1) * 15 = 1 * 15 = 15
+  // Page = 3: (3 - 1) * 15 = 2 * 15 = 30
+  const offset = (page - 1) * limit;
+
+  return {
+    limit,
+    offset,
+    onClickNext,
+    onClickPrev,
+    page,
+  };
+};
+
+function BlissPage() {
+  const dispatch = useDispatch();
+  const status = useSelector(blissStatus);
+  const error = useSelector(blissError);
+  let content;
+  const [addBliss, setAddBliss] = useState(false);
+  const { limit, offset, onClickNext, onClickPrev, page } = usePagination();
 
-	const onPrevPageClick = async (e) => {
-		e.preventDefault();
-		setPage((page) => page - 1);
-		setqparams({ page });
-		// await dispatch(fetchBliss({limit, offset: page.current === 1 ? 0 : (page.current * limit) - 1, page}))
-	};
+  // We want to re-fetch the results whenever the limit, offset, or page changes.
+  useEffect(() => {
+    dispatch(fetchBliss({ limit, offset, page }));
```

Comment:
I'd keep this effect even simpler, and just run it whenever a parameter changes that would change the result.  It'll also run when the component first runs.